x/claim/keeper/claim_test.go

```diff
@@ -454,68 +450,125 @@ func (suite *KeeperTestSuite) TestClaimOfDecayed() {
+		InitialClaimableAmount: initClaim,
+		ActionsCompleted:       []bool{false, false, false, false},
+	}
+	suite.app.AccountKeeper.SetAccount(ctx, authtypes.NewBaseAccount(addr1, nil, 0, 1))
+	suite.app.ClaimKeeper.SetClaimRecord(ctx, addr1, claimRecord)
+	claimedCoins, err := suite.app.ClaimKeeper.ClaimCoinsForAction(ctx, addr1, types.ActionEVM)
+	suite.Require().NoError(err)
+	coins = suite.app.ClaimKeeper.GetModuleAccountBalances(ctx)
+	suite.Require().Equal(coins.AmountOf(params.GetClaimDenom()), escrow.Sub(claimedCoins))
+
+	// End the airdrop
+	suite.app.ClaimKeeper.EndAirdrop(ctx, params)
+
+	// Make sure no one can claim after airdrop ends
+	claimedCoinsAfter, _ := suite.app.ClaimKeeper.ClaimCoinsForAction(ctx, addr1, types.ActionDelegate)
+	suite.Require().Equal(claimedCoinsAfter, sdk.NewInt(0))
+
+	// ensure claim is disabled and the module account is empty
+	params = suite.app.ClaimKeeper.GetParams(ctx)
+	suite.Require().False(params.EnableClaim)
+	coins = suite.app.ClaimKeeper.GetModuleAccountBalances(ctx)
+	suite.Require().Equal(coins.AmountOf(params.GetClaimDenom()), sdk.NewInt(0))
+
+	// ensure commuity pool has the unclaimed escrow amount
+	bal = suite.app.BankKeeper.GetBalance(ctx, distrModuleAddr, params.GetClaimDenom())
+	suite.Require().Equal(bal.Amount, escrow.Sub(claimedCoins))
+
+	// make sure the claim records is empty
+	suite.Require().True(len(suite.app.ClaimKeeper.GetClaimRecords(ctx)) == 0)
+}
+
+func (suite *KeeperTestSuite) TestClawbackAccountsAirdrop() {
+	suite.SetupClaimTest()
+
+	claimRecord := types.ClaimRecord{
+		InitialClaimableAmount: sdk.NewInt(100),
+		ActionsCompleted:       []bool{false, false, false, false},
+	}
+
+	params := suite.app.ClaimKeeper.GetParams(suite.ctx)
+
+	tests := []struct {
+		name           string
+		address        string
+		sequence       uint64
+		expectClawback bool
+	}{
+		{
+			name:           "address active",
+			address:        "evmos1qql8ag4cluz6r4dz28p3w00dnc9w8ueuafmxps",
+			sequence:       1,
+			expectClawback: false,
+		},
+		{
+			name:           "address inactive",
+			address:        "evmos1x2w87cvt5mqjncav4lxy8yfreynn273xn5335v",
+			sequence:       0,
+			expectClawback: true,
+		},
+		{
+			name:           "non airdrop address active",
+			address:        sdk.AccAddress(tests.GenerateAddress().String()).String(),
+			sequence:       1,
+			expectClawback: false,
+		},
+	}
+
+	for _, tc := range tests {
+		addr, err := sdk.AccAddressFromBech32(tc.address)
+		suite.Require().NoError(err, "err: %s test: %s", err, tc.name)
+		acc := &ethermint.EthAccount{
+			BaseAccount: authtypes.NewBaseAccount(sdk.AccAddress(addr.Bytes()), nil, 0, 0),
+			CodeHash:    common.BytesToHash(crypto.Keccak256(nil)).String(),
+		}
+		err = acc.SetSequence(tc.sequence)
+		suite.Require().NoError(err, "err: %s test: %s", err, tc.name)
+		suite.app.AccountKeeper.SetAccount(suite.ctx, acc)
+		suite.app.ClaimKeeper.SetClaimRecord(suite.ctx, addr, claimRecord)
```

Comment:
do we want to try using different claim records (i.e with different sets of actions completed)?